src/transport/SecureSession.h

```diff
@@ -170,8 +171,12 @@ class SecureSession : public Session, public ReferenceCounted<SecureSession, Sec
      */
     void MarkAsDefunct();
 
-    // Used to prevent any new exchange created on the session while the existing exchanges finish their work.
-    void MarkInactive();
+    // Used to make the session look like it's being released, while allowing
+    // the passed-in holder (which must be holding this session) to continue
```

Comment:
> (which must be holding this session)

Is it possible to have logic that enforces this?

Reply:
We crash if that's not the case.